exp/controllers/azuremachinepool_reconciler.go

```diff
@@ -52,6 +55,30 @@ func newAzureMachinePoolService(machinePoolScope *scope.MachinePoolScope) (*azur
 	}, nil
 }
 
+func (s *azureMachinePoolService) Snapshot(subscriptionID string, cred *azidentity.DefaultAzureCredential, snapshotName string, resourceGroup string, location string, ctx context.Context, osDisk *string) error {
+	snapshotFactory, err := armcompute.NewSnapshotsClient(subscriptionID, cred, nil)
+
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create snapshot client")
+	}
+
+	_, err = snapshotFactory.BeginCreateOrUpdate(ctx, resourceGroup, snapshotName, armcompute.Snapshot{ // step 3
```

Comment:
I imagine we can't reasonably throw away this first return value in order to check the ultimate result of the operation, beyond only if there were no errors beginning the operation which this already checks.